atom/browser/api/atom_api_net_log.cc

```diff
@@ -40,22 +48,58 @@ void NetLog::StartLogging(mate::Arguments* args) {
     return;
   }
 
-  net_log_->StartDynamicLogging(log_path);
+  // TODO(deepak1556): Provide more flexibility to this module
+  // by allowing customizations on the capturing options.
+  net_log_writer_->StartNetLog(
+      log_path, net::NetLogCaptureMode::Default(),
+      net_log::NetExportFileWriter::kNoLimit /* file size limit */,
+      base::CommandLine::ForCurrentProcess()->GetCommandLineString(),
+      std::string(), {} /* URLRequestContextGetter list */);
 }
 
 bool NetLog::IsCurrentlyLogging() {
-  return net_log_->IsDynamicLogging();
+  const base::Value* current_log_state =
+      net_log_state_->FindKeyOfType("state", base::Value::Type::STRING);
+  if (current_log_state && current_log_state->GetString() == "LOGGING")
+    return true;
+
+  return false;
 }
 
-base::FilePath::StringType NetLog::GetCurrentlyLoggingPath() {
-  return net_log_->GetDynamicLoggingPath().value();
+std::string NetLog::GetCurrentlyLoggingPath() {
+  // Net log exporter has a default path which will be used
+  // when no log path is provided, but since we don't allow
+  // net log capture without user provided file path, this
+  // check is completely safe.
+  if (IsCurrentlyLogging()) {
+    const base::Value* current_log_path =
+        net_log_state_->FindKeyOfType("file", base::Value::Type::STRING);
+    if (current_log_path)
+      return current_log_path->GetString();
+  }
+
+  return std::string();
 }
 
 void NetLog::StopLogging(mate::Arguments* args) {
-  base::OnceClosure callback;
-  args->GetNext(&callback);
+  args->GetNext(&stop_callback_);
```

Comment:
if this gets called twice quickly, then only one of the callbacks will get called.

Reply:
Have updated to queue up the callbacks when the logging is not stopped https://github.com/electron/electron/pull/14656/commits/5fa25ade920550e9b3cb06838639103237916a65